### DISCUSSION TITLE
Eglot inlay hints should "shift over" display-fill-column-indicator

### DISCUSSION THREAD

**Discussion by ArsenArsen** [joaotavora/eglot#1428] [Ideas]
Hi!

Currently, when writing code which has inlay hints in it I find myself often looking for the column number indicator because I cannot trust the `display-fill-column-indicator` when inlay hints are present.  For example, in the following:

![image](https://github.com/user-attachments/assets/3f12a36c-70dd-4cc2-a5af-34d22dff619b)

... visually, it appears like this line has crossed the fill-column, and should be broken, but that is not the case: we are on column 71 (fill-column is 79).

Can the inlay hints somehow 'push' this indicator along with them on this line specifically?  I realize this'd give the appearance of a broken up line, which isn't the most visually appealing, but I think I'd benefit from that information.

I'm also open to other alternatives for preserving this information.

TIA, have a lovely day!

**Comment by joaotavora** [comment]
I don't think it will be easy without display support : the inlay hints annotations are just overlay `before-string` and `after-string`.  The problem can strike anywhere this feature is used.  I suggest you report this to bug-gnu-emacs@gnu.org

**Reply by ArsenArsen** [reply]
Okay, thank you - will do later today.  And thanks for maintaining Eglot :-)

**Reply by joaotavora** [reply]
I think `whitespace-mode` does what you want, btw:

![image](https://github.com/user-attachments/assets/2d103cee-3d01-4159-b8e3-a24ec3d78e19)

**Reply by ArsenArsen** [reply]
Hm, yes, that seems good also, thank you!  Might be better than my idea also since there's no jagged lines.
